### ISSUE TITLE
Code refactoring...?

### ISSUE THREAD

**Issue by samcunliffe** [UCL/dxss#40]
Possibilities that seem sane to us: 
* Strategy pattern.
* Factory pattern.
* (MVC...? I'm less convinced of this).

## Blocked by

* #39 

## Potential subtasks

- [ ] #41
- [ ] #42

**Comment by samcunliffe** [comment]
Closing as this is a bit out of scope given our available time.
